### DISCUSSION TITLE
Erro ao usar a lib

### DISCUSSION THREAD

**Discussion by PauloKoglin** [Menighin/cei-crawler#30] [General]
Olá, 

Importei a lib para fazer um teste básico do seu funcionamento, porém estou obtendo um erro ao importar a lib. Alguma ideia?

Versão do node: v10.16.0

Meu código:

index.js
```
const CeiCrawler = require("cei-crawler");

console.log("hallo world");
```

O erro:
```

C:\Git\Github\vs-code-ws\cei-test\node_modules\cei-crawler\src\lib\CeiCrawler.js:13
    _isLogged = false;
              ^

SyntaxError: Unexpected token =
    at Module._compile (internal/modules/cjs/loader.js:721:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (C:\Git\Github\vs-code-ws\cei-test\node_modules\cei-crawler\src\app.js:1:20)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
```

**Comment by felipesdias** [comment]
Você esta usando algum transpiler ?

**Comment by Menighin** [comment]
Olá @PauloKoglin ,
A lib usa `private class fields` que é uma feature do Node 12+.
Da uma olhada nessa issue: https://github.com/Menighin/cei-crawler/issues/1

Lembrando que a versão 14 é a LTS corrente.
Abs,

**Reply by PauloKoglin** [reply]
Olá @Menighin, 

Obrigado pelo retorno, realmente o problema era a versão do meu node.
Instalei a versão mais recente e o problema não ocorreu mais.

Aproveitando o gancho...suponho que você utilize a lib em algum projeto que consolide a carteira de investimentos ou afins.
A B3 costuma alterar muito a página do CEI a ponto de quebrar o algoritmo?

Estou procurando uma forma (segura e estável) de obter as infos da carteira, mas me parece que ninguém oferece um API diretamente integrada com a B3. A própria B3, até onde sei, também não possui APIs. 

Abraço

**Reply by Menighin** [reply]
Eu utilizo a lib no [Stoincs](https://www.stoincs.com.br/).
Ja faz alguns bons meses que utilizo ela, quase um ano acredito.
Tivemos um susto com a questão da Captcha mas o CEI disponibilizou um link que nao valida...
Nesse tempo todo nao tivemos alteração que impactasse nao, mas não é de fato nada garantido. Amanhã ou depois podem quebrar tudo e já era.

Já existe um movimento por parte da B3 de disponibilizar uma API e matar o CEI: http://www.b3.com.br/pt_br/noticias/open-investment.htm

Porém isso nao ta nada muito claro... Eu tentei contato com o pessoal e nao obtive muita resposta...
É realmente uma caixa preta.

Abraço,

**Reply by PauloKoglin** [reply]
Legal esse app, é de sua autoria?

Pois então, também já entramos em contato com a B3, na ocasião para saber se havia uma API para obter as cotação, mas eles somente nos encaminharam para os "vendors", ou seja, teria de pôr a mão no bolso. Apenas comentaram que estavam desenvolvendo APIs, mas não especificaram quais e para quando. Tempo depois descobri esse link que você comentou ai, mas outra vez não diz muito a respeito de prazos e como será a comercialização.

Me pergunto como apps como Kinvo, TradeMap, etc obtém essas informações. Não acredito que seja através de webscrapping, porque há planos pagos nessas plataformas e seria muito arriscado depender somente da boa vontade da B3 não alterar o CEI.

**Reply by felipesdias** [reply]
Eu não duvidaria destes grandes apps fazerem scrapping não, ao menos para os serviços gratuitos. Lembro que na época em que colocaram o CAPTCHA alguns deles ficaram sem funcionar a importação por um tempo

**Reply by PauloKoglin** [reply]
@felipesdias, não sabia tiveram problamas com o CAPTCHA, interessante. Isso realmente diz muita coisa. Acho pouco provável que eles tenham uma forma de importar os dados para o "free" e outra para os "pagos". Neste caso realmente pode ser que se arrisquem depender da estabilidade do scrapper. 

Até imagino que o fato que o @Menighin comentou, que eles disponibilizaram um link sem CAPTCHA pode ter sido pressão desses "grandes" que devem ter networking lá dentro, vide Kinvo por exemplo, que tem um pezinho do Tiago Nigro lá dentro, que por sua vez tem a XP.

**Reply by felipesdias** [reply]
@PauloKoglin realmente, deve ter rolado uma pressão mesmo hahah.

Outra coisa que observei, pode ter sido coincidência, mas depois que fizemos as melhorias no crawler para não usar mais o pupperter vi que o serviço de integração de alguns apps melhoraram também hahah

**Reply by PauloKoglin** [reply]
@felipesdias não duvido que eles usem essa lib também. 😄 
Bom, então torçamos que a B3 não faça nenhum movimento brusco, se fizer, que ao menos entregue as tão esperadas APIs antes.
